### PR TITLE
Three fixes for accidental breakages in recent changes

### PR DESCRIPTION
**Pull request by hulpke**
Fix for #3868: Force maximal degree reduction of automorphism group of fitting free (if needed)

Redid fix for #3864:

The previous fix (#3865) created a homomorphism object that is expensive to create for matrix groups. (matgrp package , https://github.com/hulpke/matgrp/issues/12 )

One thus must be more careful in making this map.

(Test is with matgrp package)

Fixed construction of simple groups whose name string includes a +/- 

All problems were only in dev, never in release versions

### CODE REVIEW

File: lib/morpheus.gi [hunk 1]
```diff
@@ -2374,8 +2374,20 @@ InstallGlobalFunction(AutomorphismGroupFittingFree,function(g)
 
   aup:=Normalizer(d,Image(emb,g));
 
-  #reduce degree
-  s:=SmallerDegreePermutationRepresentation(aup:cheap);
+  #reduce degree -- avoid redundatnt fixedpoints
```
Comment (fingolfin) [inline lib/morpheus.gi]:
```suggestion
  #reduce degree -- avoid redundant fixed points
```

File: lib/morpheus.gi [hunk 2]
```diff
@@ -2374,8 +2374,20 @@ InstallGlobalFunction(AutomorphismGroupFittingFree,function(g)
...
+  if count>1 then
+    p:=aup;
+    s:=IdentityMapping(p);
+    repeat
+      i:=SmallerDegreePermutationRepresentation(p);
+      if NrMovedPoints(Image(i))<NrMovedPoints(p) then
+        p:=Image(i,p);
```
Comment (fingolfin) [inline lib/morpheus.gi]: Perhaps like this for consistency?
```suggestion
        p:=Image(i);
```
Could also use `ImagesSource(i)` here and in the line above

File: lib/morpheus.gi [hunk 3]
```diff
@@ -2395,13 +2407,14 @@ InstallGlobalFunction(AutomorphismGroupFittingFree,function(g)
...
 
-  # produce data to map fro au to aup:
+  # produce data to map from au to aup:
   lemb:=MovedPoints(aup);
   stbs:=[];
   orb:=Orbits(aup,MovedPoints(aup));
   base:=BaseStabChain(StabChainMutable(aup));
   newbas:=[];
   for i in orb do
+
```
Comment (fingolfin) [inline lib/morpheus.gi]:
```suggestion
```

File: lib/morpheus.gi (hunk 2 above)
Comment (hulpke) [reply lib/morpheus.gi]: The `Image(i,p)` was deliberate, to ensure that -- even if the homomorphism is defined on a larger set, which I cannot prove will never happen -- the image group is the same. But the first `Image` could be the range.

File: grp/simple.gi [hunk 4]
```diff
@@ -329,15 +329,15 @@ local brg,str,p,a,param,g,s,small,plus,sets;
...
-      if a[p+1]='1' and a[p+2]=',' then
+      elif Length(a)>=p+1 and a[p+1]='1' and a[p+2]=',' then
 	# gave O(+1,8,2) or so
 	plus:=fail;
       fi;
       if plus<>fail then
 	if p=1 then
 	  # leading +-, possibly with comma
 	  a:=a{[2..Length(a)]};
+          if a="1" then a:="";fi;
```
Comment (fingolfin) [inline grp/simple.gi]: odd indentation (not important)

**Comment by hulpke** [comment]
Added test for the broken case (which only involved the O+/- cases). The syntax for other classes is alreday tested through the simple groups iterator.

File: grp/simple.gi (hunk 4 above)
Comment (hulpke) [reply grp/simple.gi]: Seems to be a tab issue. Will change

Comment (fingolfin) [reply grp/simple.gi]: Thanks

**Comment by fingolfin** [comment]
@hulpke in order to backport this, I'll have to backport PR #3790 as well (the alternative would be to untangle the code, but that doesn't seem to be worth the effort)

**Comment by fingolfin** [comment]
Backported to stable-4.11 in commit fdc4f1b9cca052406804ac9af2037e8ad9539b97

**Comment by hulpke** [comment]
@fingolfin Sorry -- I had added it because I though #3790 had been backported already. Thanks in any case.

**Pull request merged**